비교결과 페이지-투자하기 버튼 하드코딩

taeyeonkim94 opened the pull request:
## 이슈 번호

- close https://github.com/2-ViewMyStartup-2-FE/2-ViewMyStartup-2team-FE/issues/19

## 작업 사항
- [x] 버튼 클릭시에 모달창이 나오게됨
- [x] 모달창의 확인버튼을 누를시에 모달창이 꺼지면서 팝업창 활성화
- [x] 팝업창 확인버튼이나 X버튼을 누를시에 꺼짐
- [x] 비밀번호입력란 토글버튼 구현

추가적 사항
- [x] compare-result 페이지 Nav와 겹치는문제 해결
- [x] media 단위를 rem에서 px로 변경
- [x] 브라우저의 창이 작아질 경우에 모달창에 스크롤이 생성됨
- [x] textarea를 사용자가 크기를 임의로 수정하지 못하게 변경
- [x]  x 버튼의 이미지를 가운데 정렬

## 기타
netlify : https://66fa5bb0c87c66094502ad0a--symphonious-macaron-816638.netlify.app/compare-result

taeyeonkim94 committed:
Implement: Hardcode modal button and popup in CompareResult page
Add Netlify configuration for redirects and SPA support

taeyeonkim94 committed:
Fix: resolve confilct

taeyeonkim94 committed:
Fix: Ensure form submission occurs when the submit button is clicked

taeyeonkim94 committed:
Merge branch 'develop' into feature/#19-compareResultInvestBtn

taeyeonkim94 committed:
Fix: Ensure form submission occurs when the submit button is clicked

taeyeonkim94 committed:
Merge branch 'feature/#19-compareResultInvestBtn' of https://github.com/2-ViewMyStartup-2-FE/2-ViewMyStartup-2team-FE into feature/#19-compareResultInvestBtn

shinji530 commented on src/component/InvestmentForm.js:
```diff
@@ -0,0 +1,86 @@
+    case "name": //type에 따라 클래스, input태그의 type, 메시지 설정
+      return {
+        frame: styles.other,
+        label: "투자자 이름",
+        inputType: "text",
+        message: "투자자 이름을 입력해 주세요"
+      };
+    case "amount":
+      return {
+        frame: styles.other,
+        label: "투자 금액",
+        inputType: "text",
+        message: "투자 금액을 입력해 주세요"
+      };
+    case "comment":
+      return {
+        frame: styles.comment,
+        label: "투자 코멘트",
+        inputType: "text",
+        message: "투자에 대한 코멘트를 입력해 주세요"
+      };
+    case "password":
+    case "confirm":
+      return {
+        frame: styles.other,
+        label: type === "password" ? "비밀번호" : "비밀번호 확인",
+        inputType: isVisible ? "text" : "password",
+        message:
+          type === "password"
+            ? "비밀번호를 입력해주세요"
+            : "비밀번호를 다시 한 번 입력해주세요"
+      };
+    default:
+      return {
+        frame: styles.other,
+        label: "기본 필드",
+        inputType: "text",
+        message: "기본 메시지를 입력하세요"
+      };
+  }
```
투자 금액에는 숫자만 들어가야 할 거 같습니다. 
또한 비밀번호가 다시 입력한 비밀번호와 일치하지 않는데도 넘어가지는데 아직 이쪽은 구현이 안된걸까요?

taeyeonkim94 replied:
네 일단 이번이슈에서는 유효성 검증을 뺀채로 구현만 한 상태입니다.
투자금액에 숫자만 들어가야되는 부분은
https://github.com/2-ViewMyStartup-2-FE/2-ViewMyStartup-2team-FE/issues/31 PR에 적용되어있습니다.
이번 이슈에서 유효성검증까지 마칠까요?

shinji530 replied:
아 그럼 31에서 다시 확인하겠습니다 이번 이슈에 유효성검증까지 추가할 필요는 없을 거 같습니다!

shinji530 commented on src/css/InvestModal.module.css:
```diff
@@ -0,0 +1,52 @@
+    justify-content: space-between;
+}
+
+@media(max-width:74.3rem) {
+    .frame {
+        width: 34.2rem;
+        height: 74.2rem;
+    }
+
+    .form {
+        width: 31.1rem;
+        height: 71rem;
+    }
+
+    .layoutContainer {
+        height: 11.3rem;
+    }
+}
```
모달 반응형이 netflify에서는 제대로 안이루어지고 있는 것 같습니다,,!

shinji530 replied:
크기는 작아지는 거 같은데 모달에서 스크롤이 이루어지지 않아서 잘리는 것 같습니다

taeyeonkim94 replied:
Y축이 작아졌을때 스크롤 구현이 필요한 상황일까요?

shinji530 replied:
넵 모달 띄운 상태에서 화면 확대해보시면 스크롤이 없어서 비밀번호 확인이나 투자하기 버튼을 누를 수 없는 문제가 있습니다

shinji530 commented on src/component/InvestmentForm.js:
```diff
@@ -0,0 +1,86 @@
+        message: "투자 금액을 입력해 주세요"
+      };
+    case "comment":
+      return {
+        frame: styles.comment,
+        label: "투자 코멘트",
+        inputType: "text",
+        message: "투자에 대한 코멘트를 입력해 주세요"
+      };
```
투자 코멘트 부분 css에서 resize 안되도록 하는 게 더 좋을 거 같습니다

taeyeonkim94 replied:
구체적으로 무슨 문제가 있어서 어떻게 하면 좋을까요?

shinji530 replied:
아 단순하게
![image](https://github.com/user-attachments/assets/4a11ec75-d009-465f-9e3f-3a62ac5ced0d)
이런식으로 보이는 경우가 생기면 보기에 안 좋을 거 같아서 말한거였습니다!
그냥 css에서 resize:none으로 아예 사이즈 변경이 불가하도록 하거나
resize: vertical; / 상하만 가능 /
이런식으로 상하로만 사이즈 변경 가능하도록 하는 게 좋을 거 같습니당,,

taeyeonkim94 committed:
1. Changed media units from rem to px
2. Added scroll to modal when browser window is resized
3. Disabled resizing for textarea
4. Centered the close button image

shinji530 reviewed (approved):
수고하셨습니다!
